bpf/bpf_sock.c

```diff
@@ -446,6 +511,11 @@ __sock4_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)
 __section("connect4")
 int sock4_connect(struct bpf_sock_addr *ctx)
 {
+	if (sock_has_magic_cookie(ctx)) {
+		per_agent_hostns_set_cookie(ctx);
```

Comment:
This will be executed on every `connect` call that `bpf_sock` intercepts for workloads, etc. AFAIU, I don't think that's the intention of @borkmann 's proposal. Netns cookies [remain constant](https://man7.org/linux/man-pages/man7/BPF-HELPERS.7.html) for their lifetime so this should be a one-time operation?

@borkmann I wonder if this workflow could work -  Agent creates a dummy socket, and passes down its socket cookie to `bpf_sock_dummy` program. It then makes a socket call that'll be intercepted and matched by the dummy program. The program then gets and stores (either writes to a map, or passes it down to the agent via ring buffer) the netns cookie (aka the "host netns" cookie) of this socket. The agent will then pass it to the `bpf_sock` program while loading. 
I think this is quite similar to your proposal, but I didn't quite understand couple of details around the ordering and such in your proposal.

Reply:
Good point! I will change the code by implementing a one-time operation. Thanks ;)